src/aws-api-mcp-server/awslabs/aws_api_mcp_server/core/agent_scripts/registry/cloudtrail-multi-region-setup.script.md

```diff
@@ -0,0 +1,246 @@
+## Parameters
+
+- **trail_name** (required): Name for the CloudTrail trail (e.g., "organization-trail", "security-audit-trail")
+- **s3_bucket_name** (required): S3 bucket name for storing CloudTrail logs (must be globally unique)
+- **region** (required): AWS region for CloudTrail and CloudWatch resources (e.g., "us-east-1", "eu-west-1")
+- **cloudwatch_log_group** (optional, default: "CloudTrail/APILogs"): CloudWatch log group name for real-time analysis
+- **enable_data_events** (optional, default: false): Enable data events for S3 and Lambda (increases costs)
+- **enable_insights** (optional, default: true): Enable CloudTrail Insights for anomaly detection
+
+## Steps
+
+### 1. Verify Dependencies
+
+Check for required tools and warn the user if any are missing.
+
+**Constraints:**
+- You MUST verify the following tools are available in your context:
+  - call_aws
+- You MUST ONLY check for tool existence and MUST NOT attempt to run the tools because running tools during verification could cause unintended side effects, consume resources unnecessarily, or trigger actions before the user is ready
+- You MUST inform the user about any missing tools with a clear message
+- You MUST ask if the user wants to proceed anyway despite missing tools
+- You MUST respect the user's decision to proceed or abort
+
+### 2. Create S3 Bucket for CloudTrail Logs
+
+Create a dedicated S3 bucket with proper permissions for CloudTrail log storage.
+
+**Constraints:**
+- You MUST create the S3 bucket using: `aws s3api create-bucket --bucket ${s3_bucket_name} --region ${region}`
+- You MUST enable versioning: `aws s3api put-bucket-versioning --bucket ${s3_bucket_name} --versioning-configuration Status=Enabled --region ${region}`
+- You MUST create and apply the CloudTrail bucket policy to allow CloudTrail service access
+- You MUST handle bucket creation errors gracefully (bucket may already exist)
+- You MUST verify bucket creation was successful before proceeding
+
+### 3. Create CloudWatch Log Group
+
+Set up CloudWatch log group for real-time log analysis.
+
+**Constraints:**
+- You MUST create the log group using: `aws logs create-log-group --log-group-name ${cloudwatch_log_group} --region ${region}`
+- You MUST set retention policy: `aws logs put-retention-policy --log-group-name ${cloudwatch_log_group} --retention-in-days 90 --region ${region}`
+- You MUST handle log group creation errors (may already exist)
+- You MUST create IAM role for CloudTrail to write to CloudWatch Logs
+
+### 4. Create IAM Role for CloudTrail
+
+Create IAM role with necessary permissions for CloudTrail operations.
+
+**Constraints:**
+- You MUST create IAM role for CloudTrail service
+- You MUST attach policies for S3 bucket access and CloudWatch Logs access
+- You MUST use least privilege principle for permissions
+- You MUST save the role ARN for trail configuration
+
+### 5. Enable Multi-Region CloudTrail
+
+Create and configure CloudTrail to capture events across all regions.
```

Comment:
In my prior experiments models have been struggling to perform certain action across all regions. Have you verified that it has indeed done this in every region? I also do not see a step to list available regions, so some opt-in regions might be skipped?